server/src/integrations/environment/environment.validation.ts

```diff
@@ -38,34 +39,44 @@ export class EnvironmentVariables {
   @IsUrl({ require_tld: false })
   FRONT_AUTH_CALLBACK_URL: string;
 
-  @IsString()
+  @Transform(({ value }) => value.toLowerCase() === 'true')
   @IsOptional()
+  @IsBoolean()
+  AUTH_GOOGLE_ENABLED?: boolean;
+
+  @IsString()
+  @ValidateIf((env) => env.AUTH_GOOGLE_ENABLED === true)
   AUTH_GOOGLE_CLIENT_ID?: string;
 
   @IsString()
-  @IsOptional()
+  @ValidateIf((env) => env.AUTH_GOOGLE_ENABLED === true)
   AUTH_GOOGLE_CLIENT_SECRET?: string;
 
   @IsUrl({ require_tld: false })
-  @IsOptional()
+  @ValidateIf((env) => env.AUTH_GOOGLE_ENABLED === true)
   AUTH_GOOGLE_CALLBACK_URL?: string;
 
   // Storage
   @IsEnum(StorageType)
   @IsOptional()
   STORAGE_TYPE?: StorageType;
 
-  @ValidateIf((_, value) => value === StorageType.S3)
+  @ValidateIf((env) => env.STORAGE_TYPE === StorageType.S3)
   @IsAWSRegion()
-  STORAGE_REGION?: AwsRegion;
+  STORAGE_S3_REGION?: AwsRegion;
+
+  @ValidateIf((env) => env.STORAGE_TYPE === StorageType.S3)
+  @IsString()
+  STORAGE_S3_NAME?: string;
 
   @IsString()
-  STORAGE_LOCATION: string;
+  @ValidateIf((env) => env.STORAGE_TYPE === StorageType.Local)
+  STORAGE_LOCAL_PATH?: string;
 }
 
 export function validate(config: Record<string, unknown>) {
   const validatedConfig = plainToClass(EnvironmentVariables, config, {
-    enableImplicitConversion: true,
+    enableImplicitConversion: false,
```

Comment:
I have disabled this because it was casting ENV_VARIABLE=false to TRUE...

Reply:
see my other comment below

Reply:
Indeed, maybe some ideas here: https://github.com/typestack/class-transformer/issues/550

Reply:
nice!

Reply:
I prefer to use `Transform` instead of disabling this, because it's going to stop inferring types on all other env variables